NetworkServer/src/geekbrains/java2/client/ClientHandler.java

```diff
@@ -0,0 +1,210 @@
+        }
+    }
+
+    private void readMessages() throws IOException{
+        while (true){
+           Command command = readCommand();
+            if(command == null){
+                continue;
+            }
+            switch (command.getType()){
+                case END:
+                    System.out.println("Received 'END' command" );
+                    return;
+                case PRIVATE_MESSAGE:{
+                    PrivateMessageCommand commandData = (PrivateMessageCommand) command.getData();
+                    String receiver = commandData.getReceiver();
+                    String message = commandData.getMessage();
+                    networkServer.sendMessage(receiver, Command.messageCommand(nickname, message));
+                    break;
+                }
+                case BROADCAST_MESSAGE:{
+                    BroadcastMessageCommand commandData = (BroadcastMessageCommand) command.getData();
+                    String message = commandData.getMessage();
+                    networkServer.broadcastMessage(Command.messageCommand(nickname, message), this);
+                    break;
+                }
+                default:
+                    System.err.println("Unknown type of command: " + command.getType());
+            }
+//            String message = in.readUTF();
+//            System.out.printf("От %s: %s%n", nickname, message);
+//            if("/end".equals(message)){
+//                return;
+//            }
+//            networkServer.broadcastMessage(nickname + ": " + message, this);
+        }
+    }
+
+    private Command readCommand()throws IOException{
+        try {
+            return (Command) in.readObject();
+        }catch (ClassNotFoundException e){
+            String errorMessage = "Unknown type of object from client!";
+            System.err.println(errorMessage);
+            e.printStackTrace();
+            sendMessage(Command.errorCommand(errorMessage));
+            return null;
+        }
+    }
+
+//    public static boolean taskComplete = false;
+////    TimeOut timeOut = new TimeOut();
+////    Thread timeOutThread = new Thread();
+//
+//    public static class TimeOut implements Runnable{
+//        public void run() {
+//            try {
+//                Thread.sleep(10000);
+//            } catch (InterruptedException e) {
+//                e.printStackTrace();
+//            }
+//            if(taskComplete ==false) {
+//                System.out.println("Timed Out");
+//                return;
+//            }
+//
+//        }
+//
+//    }
+//    private void authenticationTime() throws IOException {
+//        TimeOut timeOut = new TimeOut();
+//        Thread timeOutThread = new Thread(timeOutThread);
+//        timeOutThread.start();
+//        authentication();
+//        taskComplete =true;
+//        while(true) {//do all other stuff }
+//        }
+//    }
+
+    private void authentication() throws IOException{
+
```

Comment:
Самый простой вариант с timeout:
```suggestion
Thread t = new Thread(() -> {
try {
    Thread.sleep(TIMEOUT_TIME);
    closeConnection();
 } catch(InterruptException) {
  System.out.println("Auth has been successful");  
 }
}).start();
```
и вызывать интерапт у данного потока в случае успешной аутентификации